examples/lock-app/cc13x2x7_26x2x7/main/AppTask.cpp

```diff
@@ -151,6 +151,27 @@ int AppTask::Init()
             ;
     }
 
+#if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING && CHIP_ENABLE_ROTATING_DEVICE_ID
+    {
+	uint8_t EUI64[8];
+        char stringEUI64[sizeof(EUI64) * 2 + 1]; // just big enough for string(EUI64)
+
+        // set serial number to flash
+        // this is before ZCL is started, so Basic Cluster populates Serial Number with this EUI64 (in string)
+        ThreadStackMgrImpl().GetIeeeEui64(EUI64);
+        sprintf(stringEUI64, "%02X%02X%02X%02X%02X%02X%02X%02X",
+			EUI64[0], EUI64[1], EUI64[2], EUI64[3], EUI64[4], EUI64[5], EUI64[6], EUI64[7]);
+	ret = ConfigurationMgr().StoreSerialNumber(stringEUI64, strlen(stringEUI64));
```

Comment:
Why is this dependent on rotating device id bits?   The attribute is enabled unconditionally....

Reply:
Correct, there is always a SerialNumber which is used (either one which has been stored in NV via KVS, or the hard-coded testing on "TEST_SN").

In this case, we are storing some other string as SerialNumber (vs "TEST_SN").

While this is not strictly dependent on rotating device id bits, when I worked on enabling this feature, it was in the context of enabling both CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING and CHIP_ENABLE_ROTATING_DEVICE_ID.

It is viable to store "stringEUI64" independently of these CHIP_ENABLE* bits, but I wanted to leave this up to the developer to decide. 
@bzbarsky-apple , do we think it is better practice to remove the CHIP_ENABLE* guards in this case?

Reply:
It's mostly a matter of what behavior we are aiming for here...  Either it makes sense to use the Thread EUI64 as the serial number, or it does not, and it doesn't seem like that should depend on the rotating device id bits...

Reply:
Summary: I will remove these macro guards for now, as there seems to be no strict dependency on them.

The behavior I was aiming for, when these two macros are enabled, is to ensure a unique value for SerialNumber (IEEE 802.15.4 EUI64 should be unique per device).
When either macro is disabled, any value could be used -- I believe "TEST_SN" is the default. I wasn't sure whether any other test flow is assuming "TEST_SN", but I did know that at least Rotating Device ID feature is disabled by default (so results should be unchanged for any test flow which has rotating device ID disabled).

From my own simple tests of commissioning and cluster control, it seems that these two have no assumption of "TEST_SN" as SerialNumber, so I am happy to remove these macro guards. We can re-consider them in the future on an as-needed basis.

Reply:
Update here: https://github.com/project-chip/connectedhomeip/pull/15367/commits/63fa0edb1f455eecc5b58090d3593308e9e220e7